e2eAIOK/deltatuner/README.md

```diff
@@ -0,0 +1,94 @@
+# Deltatuner
+Deltatuner is an extension for [Peft](https://github.com/huggingface/peft) to improve LLM fine-tuning speed through multiple optimizations, including: leverage the compact model constructor [DE-NAS](https://github.com/intel/e2eAIOK/tree/main/e2eAIOK/DeNas) to construct/modify the compact delta layers in a hardware-aware and train-free approach, and adding more new deltatuning algorithms.
+
+## Introduction
+<p align="center">
+  <img width="90%" src="./doc/deltatuner.png">
+</p>
+
+### Key Components
+- Supported parameter efficient finetuning algorithms
+  - [LoRA](https://arxiv.org/pdf/2106.09685.pdf) algorithm
+  - Scaling and Shifting([SSF](https://arxiv.org/abs/2210.08823)) algorithm: Scale and Shift the deep features in a pre-trained model with much less parameters to catch up with the performance of full finetuning
+  - WIP on adding more algos (AdaLora etc.)
+- De-Nas: Automatically construct compact and optimal delta layers with train-free and hardware-aware mode (more details [here](https://community.intel.com/t5/Blogs/Tech-Innovation/Artificial-Intelligence-AI/Multi-Model-Hardware-Aware-Train-Free-Neural-Architecture-Search/post/1479863))
+  - step1: Generate search space for delta layers
+  - step2: Search algorithm populates delta layers for LM
+  - step3: Train-free score evaluates LM with adaptive delta layers
+
+### Features
+- Easy-to-use: provide package install, just need to inject few codes into the original code 
+- Auto-tuning: automatically select best algorithms and delta structure for finetuning model
+
+### Values
+- Save resources: Reduce trainable parameter / Reduce fine-tuning time / Reduce memory consumption
+- Get improved/comparable accuracy
+
+## Get Started
+
+### Installation
+- install the python package
+```shell
+pip install deltatuner
+```
+
+### Fast Fine-tuning on Base models
+Below is an example of optimizing [MPT](https://huggingface.co/mosaicml/mpt-7b) model by adding the following few-lines to use the delatuner optimizations. It use the DE-NAS in delatuner to optimize a LLM with LoRA layers to a LLM with compact delta layers, so as to efficiently improve the LLM fine-tuning process in peak memory reduction and time speedup. 
```

Comment:
what does optimize LoRA layers to compact delta layers mean?

LoRA layer -> compact delta layer?

Reply:
Complex LoRA layer -> Simple LoRA layer. 
I have changed the description to 'LoRA layers to compact LoRA layers'.